File: backend/app/schemas/document.py
import uuid
from datetime import date, datetime

from pydantic import BaseModel, ConfigDict


class DocumentBase(BaseModel):
    document_number: str
    title: str
    document_type: str
    status: str = "draft"
    is_controlled: bool = True
    review_due_date: date | None = None
    metadata_json: dict | None = None
    program_id: uuid.UUID | None = None
    owner_user_id: uuid.UUID | None = None


class DocumentCreate(DocumentBase):
    tenant_id: uuid.UUID
    created_by_user_id: uuid.UUID | None = None
    updated_by_user_id: uuid.UUID | None = None


class DocumentResponse(DocumentBase):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    tenant_id: uuid.UUID
    current_revision_id: uuid.UUID | None = None
    created_by_user_id: uuid.UUID | None = None
    updated_by_user_id: uuid.UUID | None = None
    created_at: datetime
    updated_at: datetime


class DocumentListResponse(BaseModel):
    items: list[DocumentResponse]
    total: int

File: frontend/src/types/document.ts
export type DocumentRecord = {
  id: string;
  tenant_id: string;
  program_id: string | null;
  document_number: string;
  title: string;
  document_type: string;
  owner_user_id: string | null;
  current_revision_id: string | null;
  status: string;
  is_controlled: boolean;
  review_due_date: string | null;
  metadata_json: Record<string, unknown> | null;
  created_by_user_id: string | null;
  updated_by_user_id: string | null;
  created_at: string;
  updated_at: string;
};

export type DocumentListResponse = {
  items: DocumentRecord[];
  total: number;
};